pkg/ipcache/ipcache.go

```diff
@@ -163,6 +163,16 @@ func (ipc *IPCache) getHostIPCache(ip string) (net.IP, uint8) {
 
 // GetK8sMetadata returns Kubernetes metadata for the given IP address.
 func (ipc *IPCache) GetK8sMetadata(ip string) *K8sMetadata {
+	ipc.mutex.RLock()
+	defer ipc.mutex.RUnlock()
+	if k8sMeta, ok := ipc.ipToK8sMetadata[ip]; ok {
+		return &k8sMeta
```

Comment:
Should we be returning a pointer to state covered by this mutex? Seems like `updateNamedPorts()` could modify it.

Reply:
@joestringer as far as I can see, the mutex is protecting the map, not its values. However I've double checked and it seems the value is never modified but I've made it clear in the function documentation.